plan/plan.go

```diff
@@ -232,6 +235,13 @@ func (p *baseLogicalPlan) storePlanInfo(prop *requiredProperty, info *physicalPl
 	return nil
 }
 
+func (p *baseLogicalPlan) buildKeyInfo() {
+	for _, child := range p.GetChildren() {
+		child.(LogicalPlan).buildKeyInfo()
+	}
+	p.schema.Keys = nil
```

Comment:
if len(children) == 1, let p.schema.Keys = p.children[0].schema.Keys.clone()
else let it be nil.